pyquil/quilbase.py

```diff
@@ -738,69 +738,75 @@ class SimpleInstruction(AbstractInstruction):
     Abstract class for simple instructions with no arguments.
     """
 
-    op: ClassVar[str]
+    instruction: ClassVar[quil_rs.Instruction]
 
     def out(self) -> str:
-        return self.op
+        return str(self.instruction)
 
     def __str__(self) -> str:
-        return self.out()
+        return str(self)
```

Comment:
This seems suspiciously like a recursive implementation. Seems like this should stay as `self.out()`

Reply:
Similarly worried about the circular definition in `UnaryClassicalInstruction.out` below.